controller/watcher.go

```diff
@@ -124,3 +130,38 @@ func setVaultClient(clients *hcat.ClientSet, conf *config.Config) error {
 
 	return clients.AddVault(vault)
 }
+
+func newWatcherEventHandler() events.EventHandler {
```

Comment:
I know this isn't a public method, but I think it could benefit from a unit test given the cases and branching?

It might be beneficial to pass in a logger via context for testing?